packages/ra-core/src/controller/list/useSelectAll.tsx

```diff
@@ -0,0 +1,117 @@
+import { useCallback } from 'react';
+import { useQueryClient } from '@tanstack/react-query';
+
+import { useNotify } from '../../notification';
+import { useDataProvider, UseGetListOptions } from '../../dataProvider';
+import type { FilterPayload, RaRecord, SortPayload } from '../../types';
+import type { UseRecordSelectionResult } from './useRecordSelection';
+
+/**
+ * Function hook to select all items of a list (until we reached the limit)
+ *
+ * @param {string} resource Required. The resource name
+ * @param {RecordSelectionModifiers} selectionModifiers Required, The selection modifiers
+ * @param {SortPayload} sort Optional. The sort object passed to the dataProvider
+ * @param {FilterPayload} filter Optional. The filter object passed to the dataProvider
+ * @returns {Function} onSelectAll A function to select all items of a list
+ *
+ * @example
+ *
+ * const MyButton = () => {
+ *   const [_, selectionModifiers] = useRecordSelection({ resource: 'posts' });
+ *   const onSelectAll = useSelectAll({
+ *       selectionModifiers,
+ *       sort: { field: 'title', order: 'ASC' },
+ *       filter: { title: 'foo' },
+ *   });
+ *   const handleClick = () => onSelectAll({
+ *       queryOptions: { meta: { foo: 'bar' } },
+ *       limit: 250,
+ *   });
+ *   return <Button onClick={handleClick}>Select All</Button>;
+ * };
+ */
+export const useSelectAll = ({
+    resource,
+    selectionModifiers,
+    sort,
+    filter,
+}: useSelectAllProps): ((options?: onSelectAllProps) => void) => {
+    const dataProvider = useDataProvider();
+    const queryClient = useQueryClient();
+    const notify = useNotify();
+
+    const onSelectAll = useCallback(
+        async ({ queryOptions = {}, limit = 250 }: onSelectAllProps = {}) => {
+            const { meta, onSuccess, onError } = queryOptions;
+            try {
+                const results = await queryClient.fetchQuery({
+                    queryKey: [
+                        resource,
+                        'getList',
+                        {
+                            resource,
```

Comment:
No, `resource` shouldn't be in the third param of the query key as it's already the first. Check `useGetList` to see the proper key.